pdns/nameserver.cc

```diff
@@ -291,12 +291,10 @@ void UDPNameserver::send(DNSPacket *p)
 
   fillMSGHdr(&msgh, &iov, cbuf, 0, (char*)buffer.c_str(), buffer.length(), &p->d_remote);
 
+  msgh.msg_control=NULL;
   if(p->d_anyLocal) {
     addCMsgSrcAddr(&msgh, cbuf, p->d_anyLocal.get_ptr(), 0);
   }
-  else {
-    msgh.msg_control=NULL;
-  }
   DLOG(L<<Logger::Notice<<"Sending a packet to "<< p->getRemote() <<" ("<< buffer.length()<<" octets)"<<endl);
```

Comment:
I know this PR is about auth and dnsdist, but it looks like we have the exact same construct in two places in pdns_recursor.cc, perhaps we could apply the same fix there?